### PR TITLE
smessage integration

### PR DESCRIPTION
**Pull request by Lagovas**
* each themis module tested separately
* secure messages tested in encryption and signing mode
* completed other integration suits (some cross-language tests wasn't added and that why decided to split them and add table of supported links of languages)

### CODE REVIEW

File: tests/_integration/integration_scell_seal.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+echo ".. testing secure cell, seal mode,  php <--> python"
+python ./tests/_integration/scell_seal_string_echo.py "dec" "passwd" `php -f ./tests/_integration/scell_seal_string_echo.php "enc" "passwd" "php->python seal"`
+check_result_zero
+echo ".. testing secure cell, seal mode,  python <--> php"
+php -f ./tests/_integration/scell_seal_string_echo.php "dec" "passwd" `python ./tests/_integration/scell_seal_string_echo.py "enc" "passwd" "python->php seal"`
+check_result_zero
+
+## with context
+echo ".. testing secure cell, seal mode context,  php <--> python"
+python ./tests/_integration/scell_seal_string_echo.py "dec" "passwd" `php -f ./tests/_integration/scell_seal_string_echo.php "enc" "passwd" "ruby->python seal with context" "somecontext"` "somecontext"
```
Comment (vixentael) [inline tests/_integration/integration_scell_seal.sh]: `php->python`

File: tests/_integration/integration_scell_seal.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+check_result_zero
+echo ".. testing secure cell, seal mode,  python <--> php"
+php -f ./tests/_integration/scell_seal_string_echo.php "dec" "passwd" `python ./tests/_integration/scell_seal_string_echo.py "enc" "passwd" "python->ruby seal with context" "somecontext"` "somecontext"
```
Comment (vixentael) [inline tests/_integration/integration_scell_seal.sh]: `python->php`

File: tests/_integration/integration_smessage_signing.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+echo ".. testing secure message, php <--> python"
+python ./tests/_integration/smessage_encryption.py "verify" "./tests/_integration/keys/server.priv" "./tests/_integration/keys/client.pub" `php -f ./tests/_integration/smessage_encryption.php "sign" "./tests/_integration/keys/client.priv" "./tests/_integration/keys/server.pub" "php -> python secure message"`
+check_result_zero
+
+# RUBY <--> GO
+echo ".. testing secure message, go <--> ruby"
+ruby ./tests/_integration/smessage_encryption.rb "verify" "./tests/_integration/keys/server.priv" "./tests/_integration/keys/client.pub" `go run ./tests/_integration/smessage_encryption.go "sign" "./tests/_integration/keys/client.priv" "./tests/_integration/keys/server.pub" "go -> python secure message"`
+check_result_zero
+echo ".. testing secure message, ruby <--> go"
+go run ./tests/_integration/smessage_encryption.go "verify" "./tests/_integration/keys/server.priv" "./tests/_integration/keys/client.pub" `ruby ./tests/_integration/smessage_encryption.rb "sign" "./tests/_integration/keys/client.priv" "./tests/_integration/keys/server.pub" "ruby secure message"`
```
Comment (vixentael) [inline tests/_integration/integration_smessage_signing.sh]: `ruby -> go`

File: tests/_integration/table_template.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#       | python | go | ruby | js | php
```
Comment (vixentael) [inline tests/_integration/table_template.txt]: is this really needed?

Comment (Lagovas) [reply tests/_integration/table_template.txt]: it's helped me to distribute such checklist across other scripts
but if you think it's unnecessary I will drop it

**Pull request merged**